src/Serialization/Fields.jl

```diff
@@ -48,3 +48,91 @@ function load_internal(s::DeserializerState, ::Type{gfp_fmpz_elem}, dict::Dict)
+# SimpleNumField
+function save_internal(s::SerializerState, K::SimpleNumField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K)),
+        :var => save_type_dispatch(s, String(var(K)))
+    )
+end
+ 
+function load_internal(s::DeserializerState, ::Type{<: SimpleNumField}, dict::Dict)
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    var = load_type_dispatch(s, dict[:var], check_namespace=false)
+    K, _ = NumberField(def_pol, var, cached=false)
+    
+    return K
+end
+
+################################################################################
+# FqNmodfinitefield
+function save_internal(s::SerializerState, K::FqNmodFiniteField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K))
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{FqNmodFiniteField}, dict::Dict)
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    K, _ = FiniteField(def_pol, cached=false)
+    
+    return K
+end
+
+#elements
+function save_internal(s::SerializerState, k::Union{nf_elem, fq_nmod, Hecke.NfRelElem}) 
+    K = parent(k)
+    polynomial = parent(defining_polynomial(K))(k)
+    K_dict = save_type_dispatch(s, K)
+    
+    return Dict(
+        :parent => K_dict,
+        :polynomial => save_type_dispatch(s, polynomial)
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{nf_elem, fq_nmod, Hecke.NfRelElem}},
+                       dict::Dict)
+    K = load_type_dispatch(s, dict[:parent], check_namespace=false)
+    polynomial = load_type_dispatch(s, dict[:polynomial], check_namespace=false)
+    return K(polynomial)
+end
+
+################################################################################
+# Non Simple Extension
+function save_internal(s::SerializerState, K::Union{NfAbsNS, NfRelNS})
+    def_pols = defining_polynomials(K)
+    return Dict(
+        :def_pols => save_type_dispatch(s, def_pols),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{NfAbsNS, NfRelNS}},
+                       dict::Dict)
+    def_pols = load_type_dispatch(s, dict[:def_pols], check_namespace=false)
+    K, _ = NumberField(def_pols, cached=false)
+    return K
+end
+
+#elements
+function save_internal(s::SerializerState, k::Union{NfAbsNSElem, Hecke.NfRelNSElem})
+  K = parent(k)
+  polynomial = Oscar.Hecke.data(k)
+  K_dict = save_type_dispatch(s, K)
+
+  return Dict(:parent => K_dict, :polynomial => save_type_dispatch(s, polynomial))
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{NfAbsNSElem, Hecke.NfRelNSElem}},
+                       dict::Dict)
+  K = load_type_dispatch(s, dict[:parent]; check_namespace=false)
+  polynomial = load_type_dispatch(s, dict[:polynomial]; check_namespace=false)
+  polynomial = evaluate(polynomial, gens(parent(K.pol[1])))
```

Comment:
```suggestion
  return evaluate(polynomial, gens(K))
```
Can you try if this works?

Reply:
nice, this works